### PR TITLE
Add ability to override column header

### PR DESCRIPTION
**Pull request by fantapop**
Some background here.  Column header can sort of be overridden by using the customHeadRender option on the column object however if you use this, you lose a lot of out of the box functionality including sorting, adjustable column widths and column hint.  Also, customHeadRender is not applied in the responsive view so if you change the names of the columns, they'll revert when on mobile.

### CODE REVIEW

**Comment by gregnb** [comment]
More specific, this PR is an attempt to provide the following to each column:
 id, displayName

is that correct?

File: src/MUIDataTable.js [hunk 1]
```diff
@@ -268,6 +269,8 @@ class MUIDataTable extends React.Component {
 
         columnOptions = {
           name: column.name,
+          label: column.label,
```
Comment (gregnb) [inline src/MUIDataTable.js]: Lets change this to 

```
label: column.label ? column.label : column.name
```

Comment (gregnb) [reply src/MUIDataTable.js]: Also, instead let's make `label`  always defined that way we can remove all these guards we have in the other files checking if label || name.

That will require changing: https://github.com/gregnb/mui-datatables/blob/master/src/MUIDataTable.js#L296
to this:
```
        columnOptions = { ...columnOptions, name: column, label: column };

```

**Comment by gregnb** [comment]
Ok I've left some comments and the project structure overall changed so if you can make those adjustments I'll review again

**Comment by fantapop** [comment]
Great. Will do.

**Comment by fantapop** [comment]
@gregnb I've made these updates.  Take another look when you have a chance.

**Comment by gregnb** [comment]
Looks good @fantapop can you put in some unit tests to confirm that if label is provided that it renders that label and if not provided it uses name? i'll be ready to approve after

**Comment by fantapop** [comment]
Yeah. I was looking for an example test which is checking for what is
rendered. Is there one I can model it after?

**Comment by russell-ingram** [comment]
Any idea when this will be merged? Was looking for a fix for this exact issue before using mui-datatables to replace some functionality in my existing app.

**Comment by fantapop** [comment]
I’ll try to figure out some tests this evening.

**Comment by fantapop** [comment]
@gregnb added some tests here.  Let me know if there's a better way.

**Comment by fantapop** [comment]
@gregnb have you had a chance to look over this one? Thanks

**Comment by gregnb** [comment]
Sorry for the delay. Just took a look and it's great. Thanks again for your hard work! Merged

**Pull request merged**